Students/Kovacs Gellert/Zoowsome/src/javasmmr/zoowsome/controllers/MainController.java

```diff
@@ -0,0 +1,25 @@
+package javasmmr.zoowsome.controllers;
+
+import javasmmr.zoowsome.models.animals.*;
+import javasmmr.zoowsome.models.animals.Mammal;
+import javasmmr.zoowsome.services.factories.AnimalFactory;
+import javasmmr.zoowsome.services.factories.Constants;
+import javasmmr.zoowsome.services.factories.MammalFactory;
+import javasmmr.zoowsome.services.factories.SpeciesFactory;
+
+public class MainController {
+	public static void main(String[] args) throws Exception {
+		AnimalFactory abstractFactory = new AnimalFactory();
```

Comment:
An excellent question you got there. The fact of the matter is that you are creating a `Cow` from the get go. You are not creating only an animal. Even though you are writing
`Animal a1 = new Cow()`
you are actually creating a new `Cow`! However, you are only able to use the methods and attributes from the animal class. 

This is known as polymorphism. 
Imagine the following example:

`Musician musician = new Pianist();`

or

`Musician musician = new Drummer();`

Suppose `Musician` class has a method: `play()`. It doesn't matter who the musician is, if you call play method, you'll know how to play the determined instrument, base on concrete class, on this case, no matter if Pianist or Drummer.

Each concrete class, overwriting its `play()` method, can play on your own:

e.g. for `Pianist` class: `play() { playsBeethoven(); }`
or for `Drummer` class: `play() { playsDreamTheater(); }`

For detailed information, please check http://docs.oracle.com/javase/tutorial/java/IandI/polymorphism.html

It's always good to remember to use it with inheritance http://docs.oracle.com/javase/tutorial/java/IandI/subclasses.html